uber/tests/models/test_badge_funcs.py

```diff
@@ -107,20 +107,21 @@ def test_preassigned_with_new(self, session):
         session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=6))
         assert 7 == session.get_next_badge_num(c.STAFF_BADGE)
 
+    def test_preassigned_two_new(self, session):
+        session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=6))
+        session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=7))
+        assert 8 == session.get_next_badge_num(c.STAFF_BADGE)
+
     def test_preassigned_with_dirty(self, session):
         session.supporter_five.badge_type, session.supporter_five.badge_num = c.STAFF_BADGE, 6
         assert 7 == session.get_next_badge_num(c.STAFF_BADGE)
         session.expunge(session.supporter_five)
 
-    def test_preassigned_with_new_gap(self, session):
-        session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=8))
-        session.add(Attendee(badge_type=c.STAFF_BADGE, badge_num=6))
-        assert 7 == session.get_next_badge_num(c.STAFF_BADGE)
-
-    def test_preassigned_with_dirty_gap(self, session):
-        session.supporter_five.badge_type, session.supporter_five.badge_num = c.STAFF_BADGE, 8
-        session.supporter_four.badge_type, session.supporter_four.badge_num = c.STAFF_BADGE, 6
-        assert 7 == session.get_next_badge_num(c.STAFF_BADGE)
+    def test_preassigned_two_dirty(self, session):
```

Comment:
The original 'gap' unit tests never caught this error even though it's (theoretically) been here since the badge number refactor. That's because the bug affects the _third_ badge to be saved (and every badge thereafter). So we add these unit tests to make sure we don't hit this bug again in the future.